cookbooks/redis/recipes/default.rb

```diff
@@ -80,7 +80,7 @@
 
 if ['solo', 'app', 'app_master', 'util'].include?(node[:instance_role])
   instances = node[:engineyard][:environment][:instances]
-  redis_instance = (node[:instance_role][/solo/] && instances.length == 1) ? instances[0] : instances.find{|i| i[:name].to_s[/redis/]}
+  redis_instance = (node[:instance_role][/solo/] && instances.length == 1) ? instances[0] : instances.find{|i| "redis" == i[:name]}
```

Comment:
Are you saying this might pick up the `redis_slave` or `redis_backup`?

If so maybe adding some more logic to deal with that collision would be good.